libafl/src/inputs/multi.rs

```diff
@@ -5,37 +5,47 @@
 //! requires that each subcomponent be the same subtype.
 
 use alloc::{
+    borrow::Cow,
+    fmt::Debug,
     string::{String, ToString},
     vec::Vec,
 };
+use core::{hash::Hash, num::NonZero};
 
 use arrayvec::ArrayVec;
+use libafl_bolts::{
+    rands::Rand as _,
+    tuples::{Map, MappingFunctor},
+    Error, Named,
+};
 use serde::{Deserialize, Serialize};
 
-use crate::{corpus::CorpusId, inputs::Input};
+use crate::{
+    corpus::CorpusId,
+    inputs::Input,
+    mutators::{MutationResult, Mutator},
+    state::HasRand,
+};
 
 /// An input composed of multiple parts. Use in situations where subcomponents are not necessarily
 /// related, or represent distinct parts of the input.
 #[derive(Clone, Debug, Serialize, Deserialize, Hash)]
-pub struct MultipartInput<I> {
+pub struct ListInput<I> {
     parts: Vec<I>,
-    names: Vec<String>,
 }
 
-impl<I> Default for MultipartInput<I> {
+impl<I> Default for ListInput<I> {
```

Comment:
Why not keep MultipartInput as name? They are roughly equal, right?

Reply:
Ah wait I think I see what you did there

Reply:
Because there could be other inputs based on other collection types such as maps. I'd personally also like to rename MultipartInput, but kept it for now because it's a change that isn't strictly necessary.

Reply:
How would this work differently for maps? isn't multipart input alrady key-value?

Reply:
Could be tree maps or sth

Reply:
Not sure I agree